apps/hash-graph/lib/graph/src/store/postgres/query/compile.rs

```diff
@@ -341,38 +335,16 @@ impl<'c, 'p: 'c, R: PostgresRecord> SelectCompiler<'c, 'p, R> {
         }
     }
 
-    pub fn compile_path_column<'q>(&mut self, path: &'p R::QueryPath<'q>) -> AliasedColumn<'c>
+    pub fn compile_path_column<'q>(&mut self, path: &'p R::QueryPath<'q>) -> AliasedColumn
     where
         R::QueryPath<'q>: PostgresQueryPath,
     {
-        let column = match path.terminating_column() {
-            Column::DataTypes(DataTypes::Schema(Some(JsonField::JsonPath(field)))) => {
-                self.artifacts.parameters.push(field);
-                Column::DataTypes(DataTypes::Schema(Some(JsonField::JsonPathParameter(
-                    self.artifacts.parameters.len(),
-                ))))
-            }
-            Column::PropertyTypes(PropertyTypes::Schema(Some(JsonField::JsonPath(field)))) => {
-                self.artifacts.parameters.push(field);
-                Column::PropertyTypes(PropertyTypes::Schema(Some(JsonField::JsonPathParameter(
-                    self.artifacts.parameters.len(),
-                ))))
-            }
-            Column::EntityTypes(EntityTypes::Schema(Some(JsonField::JsonPath(field)))) => {
-                self.artifacts.parameters.push(field);
-                Column::EntityTypes(EntityTypes::Schema(Some(JsonField::JsonPathParameter(
-                    self.artifacts.parameters.len(),
-                ))))
-            }
-            Column::EntityEditions(EntityEditions::Properties(Some(JsonField::JsonPath(
-                field,
-            )))) => {
-                self.artifacts.parameters.push(field);
-                Column::EntityEditions(EntityEditions::Properties(Some(
-                    JsonField::JsonPathParameter(self.artifacts.parameters.len()),
-                )))
-            }
-            column => column,
+        let (column, parameter) = path
+            .terminating_column()
+            .into_owned(self.artifacts.parameters.len() + 1);
```

Comment:
Why is the `+1` required here? wouldn't the `.len()` be 1 higher than the current index automatically? I would assume that `len() + 1` would cause the current index to be one higher than the current index?

Reply:
Ah of course. Makes sense :+1: